service/api_task.go

```diff
@@ -317,7 +320,36 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
 	// continue must be preceded by dequeueing the current task from the
 	// queue to prevent an infinite loop.
 	for taskQueue.Length() != 0 {
-		queueItem := taskQueue.FindNextTask(spec)
+		var queueItem *model.TaskQueueItem
+		var err error
+		switch currentHost.Distro.PlannerSettings.Version {
+		case evergreen.PlannerVersionTunable:
+			if taskQueueService == nil {
+				taskQueueService = model.NewTaskQueueService(taskQueueServiceTTL)
+			} else {
+				if err = taskQueueService.Refresh(currentHost.Distro.Id); err != nil {
+					grip.Alert(message.WrapError(err, message.Fields{
+						"distro":  currentHost.Distro.Id,
+						"host":    currentHost.Id,
+						"message": "problem refreshing taskQueueService",
+						"spec":    spec,
+					}))
+					return nil, errors.Wrap(err, "problem refreshing taskQueueService")
+				}
+			}
+			queueItem, err = taskQueueService.FindNextTask(currentHost.Distro.Id, spec)
```

Comment:
Maybe write a wrapper (or a method somewhere) that does the "refresh if needed?" logic or just be able to call refresh and have it noop always?

Reply:
done